### PR TITLE
rust: Add v1.70.0 and simplify package bootstrap

### PR DESCRIPTION
**Pull request by alecbcs**
As is, the Rust package is rather difficult to maintain and update due to the embedded binary bootstrap compiler. (@AndrewGaspar absolutely no disrespect, you've done a lot to work on the Rust package to make it work while being a complex build.)

This PR refactors the Rust package to move the bootstrap compiler to it's own package. This allows us to utilize the concretizer to determine valid bootstrap compilers for Rust (typically the previous major version as well as the current) and results in more reuse capability for users as well as a more maintainable and cleaner package format.

### CODE REVIEW

**Comment by alecbcs** [comment]
I've added back the `@nightly` version to fix `py-tokenizers`. After this PR is merged I think it's probably worth working to remove references to nightly versions of Rust to improve reproducibility now that the Rust ecosystem is more mature.

**Comment by alecbcs** [comment]
@spackbot run pipeline

**Review by alalazo — COMMENTED** [review]
Dividing into two recipes seems a nice way to reduce complexity. Waiting for @AndrewGaspar comments on this one.

**Comment by alecbcs** [comment]
@AndrewGaspar I noticed you recently moved from LANL to Meta. If you're no longer using Spack, I could remove you as maintainer if you'd like.

**Comment by alecbcs** [comment]
@alalazo It's been about a week at this point without any word from @AndrewGaspar. Would you be willing to review in his place?

**Review by alalazo — APPROVED** [review]
LGTM if pipelines pass

**Pull request merged**
